### PR TITLE
API branch

### PR DESCRIPTION
**Pull request by Gambit142**
In this branch, I posted and got data from the [Leaderboard Api Service](https://www.notion.so/Leaderboard-API-service-24c0c3c116974ac49488d4eb0267ade3). 

The following project requirements were fulfilled:
- Create a new game with the name of your choice by using the API.
- Implement the "Refresh" button (receiving data from the API and parsing the JSON).
- Implement the form "Submit" button (sending data to the API).
- Use async and await JavaScript features to consume the API.

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const response = await fetch(gameNameUrl, {
+    method: 'POST',
+    body: JSON.stringify({ name: gameName }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  const gameData = await response.json();
+  return gameData;
+};
```
Comment (od-c0d3r) [inline src/api.js:15]: Great job implementing  `async / await` function to create a game object.. with predefined name of your choice 👍🏽

File: src/api.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const UserInputUrl = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/z7JM1rmeo6NjLkeyvLYM/scores/';
```
Comment (o-rashad) [inline src/api.js:1]: _[OPTIONAL]_ However it's not a requirement, but if you want to practice some ES6 string methods 🤠, try to retrieve the ID from the response of creating your game object using 
```javascript
gameData.result
``` 
from inside `createGameName()` funticon
then you can use string interpolation to update the input URL like
```javascript
UserInputUrl = `https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameID}/scores/`;
```

**Review by o-rashad — APPROVED** [review]
Hi @Gambit142 , 👋🏽😄 , Omar is here..

Once I entered your PR I noticed the efforts you put into it. 👏🏼
Well, descriptive commit messages with a nice PR Header. ✅
Clean code 💯 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/api.js (hunk 2 above)
Comment (Gambit142) [reply src/api.js:1]: Thanks for the suggestions and the feedback. You rock 👍

**Pull request merged**